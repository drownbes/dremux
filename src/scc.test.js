const { Graph } = require('./graph');
const { tarjanScc } = require('./scc');

describe('scc', () => {

  test('scc tests', () => {
    const g = new Graph();
    g.addEdge(0, 1);
    g.addEdge(0, 6);
    g.addEdge(1, 2);
    g.addEdge(1, 5);
    g.addEdge(2, 3);
    g.addEdge(3, 4);
    g.addEdge(4, 2);
    g.addEdge(5, 0);

    const res = tarjanScc(g);
    expect(res).toMatchSnapshot();

  });
});
