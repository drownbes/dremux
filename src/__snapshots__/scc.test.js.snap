// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scc scc tests 1`] = `
Array [
  Graph {
    "_data": Object {
      "2": Array [
        "3",
      ],
      "3": Array [
        "4",
      ],
      "4": Array [
        "2",
      ],
    },
    "vertexParams": Object {},
  },
  Graph {
    "_data": Object {
      "6": Array [],
    },
    "vertexParams": Object {},
  },
  Graph {
    "_data": Object {
      "0": Array [
        "1",
      ],
      "1": Array [
        "5",
      ],
      "5": Array [
        "0",
      ],
    },
    "vertexParams": Object {},
  },
]
`;
